### DISCUSSION TITLE
Cloud Storage Configuration for AWS in knowlg inQuiry Setup

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Ed/Community#749] [Sunbird ED configuration discussion]
Hello,

We have done the fresh setup of knowlg inQuiry setup for one of our project where we are using AWS.
Along with the knowlg inQuiry services, we are planning to use the knowlg and inQuiry editors and players.

We initially started with the [inQuiry Editor](https://github.com/Sunbird-inQuiry/editor) and successfully integrated it into our application. In the configuration for the Questionset Editor, there is a cloudStorage configuration that looks like this which is specific to azure:

```
const editorConfig = {
  context: {
   cloudStorageUrls: ['{{cloudStorageUrls path value}}'],
    cloudStorage: {
      provider: 'azure',
      presigned_headers: {
        'x-ms-blob-type': 'BlockBlob'
      }
    },
    // ...other configurations
  },
  config: {
    // ...other configurations
  }
}

```
The Questionset Editor internally uses the [@project-sunbird/sunbird-file-upload-library](https://www.npmjs.com/package/@project-sunbird/sunbird-file-upload-library)

I wanted to ask:

What should the cloudStorage configuration look like if we are using AWS?
How should the presigned_headers be structured for AWS in this case?
Any guidance or examples would be appreciated.

Thanks in advance!

@pallakartheekreddy @HarishGangula @rajeevsathish @vinukumar-vs @vaibhavbhuva
@coolbung @itsvick @AbhilashKD

**Comment by HarishGangula** [comment]
@rajnishdargan  Create the thread in Knowlg Building block
